### ISSUE TITLE
(Release): Natron 2.4.3 - any blockers?

### ISSUE THREAD

**Issue by devernay** [NatronGitHub/Natron#752]
Does anyone have blockers or things they want to merge before 2.4.3?

- ~~OCIO2 on WIndows https://github.com/NatronGitHub/Natron/issues/752#issuecomment-1011990270~~
- ~~PyPlug overlay/interact missing in 2.4.3 #762~~

For the next 2.5.0 alpha, we are waiting for the Windows build including Python 3 (I can hear it coming).

**Comment by rodlie** [comment]
> color palette?

I will add the color palette, will finish it after work today.

> For the next 2.5.0 alpha, we are waiting for the Windows build

Might need some minor adjustment to the build scripts, else should be ok. Waiting on the latest build to see if something failed.

> Does anyone have blockers

Not a blocker per se, but openfx-io does not work/load in Windows if IOSupport/glsl is built. So still on OCIO1.

**Comment by devernay** [comment]
OCIO1 is an issue because GPU rendering is wrong, and the vfx reference platform for CY2021 is OCIO2, so people may have configs that don't load anymore with OCIO1.

OCIO2 brings many new things, including better native ACES support https://www.digitalmediaworld.tv/post/3101-opencolorio-version-2-updates-aces-transforms-and-gpu-renderer

Do ociodisplay work? It should use those same helpers.

Can you please post a separate issue describing the problem more precisely?

**Comment by devernay** [comment]
The OCIO2 code (glsl.cpp) does the same OpenGL calls as the OCIO1 version (see here https://github.com/NatronGitHub/openfx-io/blob/master/IOSupport/GenericOCIOOpenGL.cpp#L103), it's just in a different source file, so I don't see a reason why it would not *load*.

Compare
```
grep "^ *gl" GenericOCIOOpenGL.cpp | sed -e 's/^ *//' |cut -f 1 -d '('|sort |uniq |fgrep -v glBuilder
```
and
```
grep "^ *gl" glsl.cpp | sed -e 's/^ *//' |cut -f 1 -d '('|sort |uniq |fgrep -v glBuilder
```

These seem to be all OpenGL 2.0 calls, so I don't see how loading could fail.

**Comment by Shrinks99** [comment]
This is specifically not a blocker but I'm trying my best to get the [new website](https://github.com/Shrinks99/NatronGitHub.github.io) done for 2.5.  Putting it on your radar because [my to-do list](https://github.com/Shrinks99/NatronGitHub.github.io/projects/1) is nearing completion!

Once this is finished the existing repo will need to be archived and renamed as  NatronGitHub.github.io-old, I will then transfer ownership of this new repo over to the org.  Any files that have been moved over to the new site have had their git history kept intact.

**Comment by devernay** [comment]
fixed blockers https://github.com/NatronGitHub/Natron/issues/756 and https://github.com/NatronGitHub/Natron/issues/755

**Comment by devernay** [comment]
@rodlie For 2.4.3 I think we also need OpenColorIO 2 on Windows to get parity of functionality between the OSes. Any update on that?

**Comment by devernay** [comment]
Can you post an issue specific to this problem? I cannot debug remotely, but I don't understand what "doesn't load" means when the symbols are not different and no OpenGL calls are done during loading.

**Comment by devernay** [comment]
I started builds for a RC1: https://github.com/NatronGitHub/Natron/releases/tag/v2.4.3-rc1

**Comment by rodlie** [comment]
Will start the Windows build in about 15min (with OCIO1).

> For 2.4.3 I think we also need OpenColorIO 2 on Windows to get parity of functionality between the OSes. Any update on that?

Ok, no update yet.

> but I don't understand what "doesn't load" means when the symbols are not different and no OpenGL calls are done during loading.

The IO OFX will not load in Natron, the same way if depends are not met. But all depends are present, readers from openfx-arena works, so I assume this is in the OCIO nodes (not readers), or whatever uses glsl?

I got tomorrow (Saturday) free, so I will try to debug this further then.

**Comment by devernay** [comment]
The strange thing is that there's absolutely no additional depend, and the OCIO1 version also uses GLSL shaders. The only difference is the use of a helper class OpenGLBuilder.
can you run a dependency walker on IO.ofx? are you sure glsl.o is linked in the binary?

**Comment by rodlie** [comment]
Got OCIO2 working (was a linker issue against OCIO in OIIO), only difference against OCIO1 build is that TestEqualize failed:

Ref:
![TestEqualize-reference1](https://user-images.githubusercontent.com/34516798/150651960-31446567-7300-4161-a0b8-05b187245f95.jpg)

Out:
![TestEqualize-output1](https://user-images.githubusercontent.com/34516798/150651967-12f1271a-8140-4f8c-8997-108f95ebc1e0.jpg)


Do you want me to replace the RC1 binary?

**Comment by devernay** [comment]
Sure!

**Comment by rodlie** [comment]
Currently doing a new build with OIIO 2.3.11, if the tests pass I will upload that instead.

**Comment by devernay** [comment]
was the build successful? If yes can you update the blockers?

**Comment by rodlie** [comment]
yes, 2.4.3 RC1 uses OCIO 2.1.1 and OIIO 2.3.11.

**Comment by devernay** [comment]
should we do a RC2 with #762 fixed, or go with the release?

**Comment by devernay** [comment]
@rodlie

**Comment by rodlie** [comment]
Sorry for the late reply.

I'm currently trying to fix the project load/save issue on Windows. So if possible at least delay a release until tomorrow (Sunday).

RC or release? we added new functionality with #766, but it's not a breaking change, so going strait to a release should not be an issue.

**Comment by devernay** [comment]
Any update, @rodlie?

**Comment by devernay** [comment]
OK let's tag a release
